app/routes.py

```diff
@@ -1,2 +1,324 @@
+
+        # if type(rental_data["customer_id"]) != int or type(rental_data["video_id"]) != int:
+        if not isinstance(rental_data["customer_id"], int) or not isinstance(rental_data["video_id"], int):
+            return {
+                "details": "Invalid ID data type."
+            }, 400
+
+        new_rental = CustomerVideoRental(customer_id=rental_data["customer_id"],
+                                        video_id=rental_data["video_id"],
+                                        due_date=datetime.datetime.now()+datetime.timedelta(days=7))
+
+        customer = Customer.query.get(new_rental.customer_id)
+        video = Video.query.get(new_rental.video_id)
+
+        if customer == None:
+            return "Customer not found.", 404
+
+        if video == None:
+            return "Video not found.", 404
+        elif video.available_inventory <= 0:
+            return {
+                "details": "0 inventory."
+            }, 400
+
+        customer.videos_checked_out_count += 1
+        video.available_inventory -= 1
+
+        db.session.add(new_rental)
+        db.session.commit()
+
+        return {
+            "customer_id": new_rental.customer_id,
+            "video_id": new_rental.video_id,
+            "due_date": new_rental.due_date,
+            "videos_checked_out_count": customer.videos_checked_out_count,
+            "available_inventory": video.available_inventory
+            }
+
+
+@rentals_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def handle_rentals_checkin():
+
+    if request.method == "POST":
+
+        checkin_data = request.get_json()
+
+        customer_checkin_id = checkin_data["customer_id"]
+        video_checkin_id = checkin_data["video_id"]
+
+        customer = Customer.query.get(customer_checkin_id)
+        video = Video.query.get(video_checkin_id)
+
+        if customer == None:
+            return "Customer not found.", 404
+
+        if video == None:
+            return "Video not found.", 404
+
+        rental = CustomerVideoRental.query.all()
+        # rental_test = CustomerVideoRental.query.filter_by(customer_id=customer_checkin_id, video_id=video_checkin_id)
+        # rental_test = CustomerVideoRental.query.filter(CustomerVideoRental.customer_id==customer_checkin_id, CustomerVideoRental.video_id==video_checkin_id)
+        
+        for record in rental:
+            if record.customer_id == customer_checkin_id and record.video_id == video_checkin_id and record.outstanding:
+                record.outstanding = False
+
+                customer.videos_checked_out_count -= 1
+                video.available_inventory += 1
+
+                db.session.commit()
+
+                return {
+                "customer_id": customer.id,
+                "video_id": video.id,
+                "videos_checked_out_count": customer.videos_checked_out_count,
+                "available_inventory": video.available_inventory
+                }
+
+        else:
```

Comment:
It looks like this tab level is incorrect. Should this be the `else` for `if record.customer_id == customer_checkin_id and record.video_id == video_checkin_id and record.outstanding:`?